src/plugins/plugin-http.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+var shimmer = require('shimmer');
+var url = require('url');
+var isString = require('is').string;
+var merge = require('lodash.merge');
+var httpAgent = require('_http_agent');
+
+function patchModuleRoot (request, api) {
+  var labels = api.labels;
+  function wrapInvocation (options, callback) {
+    return function httpActionTrace () {
+      if (!this._google_trace_patched && options) {
+        // Don't keep wrapping our same request
+        this._google_trace_patched = true;
+        return patchHTTPRequest(options, callback, request);
+      }
+      return request.apply(this, arguments);
+    }
+  }
+
+  function setTraceHeader (parsedOptions, context) {
+    if (context) {
+      return merge(parsedOptions, {
+        headers: {
```

Comment:
Does this overwrite `parsedOptions.headers` entirely? Maybe it should be `merge(parsedOptions.headers, {[TraceLabels.TRACE_CONTEXT_HEADER_NAME]: context})` instead?